test/functional/Java8andUp/src/org/openj9/test/booleans/BooleanTest.java

```diff
@@ -0,0 +1,265 @@
+	}
+	
+	@Test
+	public static void testJNINativeBoolReturnFromGeneratedByteCodes() throws Throwable {
+		testJNINativeBoolReturnWithASM(testBoolTester, 0xFF, 1);
+		testJNINativeBoolReturnWithASM(testBoolTester, 0x08, 1);
+		testJNINativeBoolReturnWithASM(testBoolTester, 0x00, 0);
+	}
+	
+	@Test
+	public static void testUnsafeGetAndPutBoolean() {
+		testUnsafePut(0xFF, 1);
+		testUnsafePut(0x08, 1);
+		testUnsafePut(0x00, 0);
+	}
+	
+	@Test
+	public static void testUnsafeGet() throws Throwable {
+		testUnsafeGet(testBoolTester, 0xFF, 1);
+		testUnsafeGet(testBoolTester, 0x08, 1);
+		testUnsafeGet(testBoolTester, 0x00, 0);
+	}
+	
+	@Test
+	public static void testUnsafeGetAndPutBooleanOnObject() {
+		TestBool instance = new TestBool();
+		testUnsafePutBoolOnObject(instance, zOffset, 0xFF, 1);
+		testUnsafePutBoolOnObject(instance, zOffset, 0x08, 1);
+		testUnsafePutBoolOnObject(instance, zOffset, 0x00, 0);
+	}
+	
+	@Test
+	public static void testPutfieldBytecode() throws Throwable {
+		TestBool instance = new TestBool();
+		testBytecodeInstancePutfield(testBoolTester, instance, 0xFF, 1);
+		testBytecodeInstancePutfield(testBoolTester, instance, 0x08, 0);
+		testBytecodeInstancePutfield(testBoolTester, instance, 0x00, 0);
+	}
+	
+	@Test
+	public static void testPutStaticfieldBytecode() throws Throwable {
+		testBytecodeStaticPutfield(testBoolTester, 0xFF, 1);
+		testBytecodeStaticPutfield(testBoolTester, 0x08, 0);
+		testBytecodeStaticPutfield(testBoolTester, 0x00, 0);
+	}
+	
+	@Test
+	public static void testReturnBooleanBytecode()  throws Throwable{
+		testReturnBoolean(testBoolTester, 0xFF, true);
+		testReturnBoolean(testBoolTester, 0x08, false);
+		testReturnBoolean(testBoolTester, 0x00, false);
+	}
+	
+	static void testJNIInstanceBoolean(TestBool instance, int value, int expected) {
+		putInstanceBoolean(TestBool.class, instance, f, value); 
+		int res = getInstanceBoolean(TestBool.class, instance, f);	
+		assertEquals(expected, res);
+	}
+	
+	static void testJNINativeBoolReturn(int value, int expected) {
+		long addr = unsafe.allocateMemory(8);
+		boolean bRes = returnBooleanValue(value);
+		unsafe.putLong(addr, 0L);
+		unsafe.putBoolean(null, addr, bRes);
+		int res = unsafe.getInt(null, addr);
+		assertEquals(expected, res);
+		unsafe.freeMemory(addr);
+	}
+	
+	static void testJNINativeBoolReturnWithASM(Class testBoolGen, int value, int expected) throws Throwable {
+		MethodHandle mh = MethodHandles.lookup().findStatic(testBoolGen, "callReturnBooleanValue", MethodType.methodType(int.class, int.class));
+		int res = (int) mh.invoke(value);
+		assertEquals(expected, res);
+	}
+
+	
+	static void testJNIStaticBoolean(int value, int expected) {
+		TestBool.zStatic = false;
+		putStaticBoolean(TestBool.class, fStatic, value); 
+		int res = getStaticBoolean(TestBool.class, fStatic);
+		assertEquals(expected, res);
+	}
+	
+	static void testUnsafePut(int value, int expected) {
+		long addr = unsafe.allocateMemory(16);
```

Comment:
Some comments would help this code.  ie: `//memset to 0`, `//write value as an int and test Unsafe's getBoolean`